### PR TITLE
[edxapp] switch to nginx-edxapp

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Instead of serving static files from a volume where we have collected them, use an `nginx` image that bundles them.

## Proposal

- [x] use `edxapp-nginx` image
- [x] remove `edxapp`'s collectstatic jobs
- [x] remove `edxapp`'s static PV

Related to #112

### CODE REVIEW

File: apps/edxapp/templates/services/nginx/configs/cms.conf.j2 [hunk 1]
```diff
@@ -59,7 +59,7 @@ server {
     gzip_types text/plain text/css text/xml text/javascript application/x-javascript application/xml image/svg+xml;
     expires {{ edxapp_nginx_cms_static_cache_expires }};
     add_header Cache-Control public;
-    root /data/static;
+    root /edx/app/edxapp/staticfiles;
```
Comment (jmaupetit) [inline apps/edxapp/templates/services/nginx/configs/cms.conf.j2:62]: should add follow symlinks here

File: apps/edxapp/templates/services/nginx/configs/lms.conf.j2 [hunk 2]
```diff
@@ -86,7 +86,7 @@ server {
     gzip_types text/plain text/css text/xml text/javascript application/x-javascript application/xml image/svg+xml;
     expires {{ edxapp_nginx_lms_static_cache_expires }};
     add_header Cache-Control public;
-    root /data/static;
+    root /edx/app/edxapp/staticfiles;
```
Comment (jmaupetit) [inline apps/edxapp/templates/services/nginx/configs/lms.conf.j2:89]: ditto

File: apps/edxapp/templates/services/nginx/configs/cms.conf.j2 (hunk 1 above)
Comment (jmaupetit) [reply apps/edxapp/templates/services/nginx/configs/cms.conf.j2:62]: According to the documentation, `nginx` follows symlinks by default: http://nginx.org/en/docs/http/ngx_http_core_module.html#disable_symlinks

**Review by sampaccoud — APPROVED** [review]

**Comment by sampaccoud** [comment]
@jmaupetit ouch I just realized that we need a custom `edxapp-nginx` image for each customer because each theme comes with custom static files! Not sure if we can make both images with a bc in openshift :thinking:

**Comment by jmaupetit** [comment]
@sampaccoud good point! I'll add an IS/BC for the edxapp/nginx service. I'll handle it in a new PR.

**Review by lunika — APPROVED** [review]

**Comment by lunika** [comment]
IMO this change is a BC break

**Comment by jmaupetit** [comment]
> IMO this change is a BC break

Yes. Should we write a migration doc?

**Comment by lunika** [comment]
We can maybe add a `UPGRADE` file like we did on some other projects like Marsha ? https://github.com/openfun/marsha/blob/master/UPGRADE.md

**Pull request merged**